obspy/io/nordic/ellipse.py

```diff
@@ -0,0 +1,329 @@
+"""
+Routines for error ellipses
+ - Calculate an ellipse from covariance matrix
+ - See if a point is inside an ellipse
+ - Calculate the angle subtended by an ellipse (for back-azimuth uncert)
+ - Plot an ellipse
+
+Still missing:
+ - ellipsoids (3D ellipses)
+ - to_uncert()  (convert ellipse to covariance matrice, needed to
+                 convert QuakeML to nordic, for example)
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import copy
+
+eps = np.finfo(float).eps
+
+
+class _ellipse:
+    def __init__(self, a, b, theta=0, center=(0, 0)):
+        """Defines an ellipse
+
+        The ellipse is assumed to be centered at zero with its semi-major axis
+        axis aligned along the NORTH axis (geographic standard, not math
+        standard!) unless orientation and/or center are set otherwise
+        You can think of it as theta (geographic angle) = 90-phi (math angle)
+
+        Inputs:
+            a=length of semi-major axis
+            b=length of semi-minor axis
+            theta=azimuth (degrees CW from 0=N (y positive))
+            center=x,y coordinates of ellipse center (0,0)
+        """
+        if a < b:
```

Comment:
If the print statement doesn't matter then this could just be:
```python
self.a = max(a, b)
self.b = min(a, b)
```
then the coverage is improved.

Reply:
I changed the print to a warnings.warn and implemented your suggestion afterwards:
```
        if a < b:
            warnings.warn('Semi-major smaller than semi-minor! Switching...')
        self.a = max(a,b)
        self.b = min(a,b)

```